replication-api-impl/src/main/java/org/codice/ditto/replication/api/impl/data/AbstractPersistable.java

```diff
@@ -92,10 +92,11 @@ private void setModified(@Nullable Date modified) {
    * before performing its own functionality.
    *
    * @param properties the properties to read into the persistable's variables
+   * @return int the version of the map passed in.
    */
-  public void fromMap(Map<String, Object> properties) {
+  public int fromMap(Map<String, Object> properties) {
     setId((String) properties.get(ID_KEY));
-    setVersion((int) properties.get(VERSION_KEY));
     setModified((Date) properties.get(MODIFIED_KEY));
+    return (int) properties.get(VERSION_KEY);
```

Comment:
✏️ Just to be safe, shouldn't we check first if the property exist and throw an `IllegalStateException` if it doesn't instead of the NPE or `ClassCastException` that would be thrown by the current code?

Reply:
Added check